File: CS/DXApplication5/AgendViewComponent/ListViewDataGenerator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraScheduler;
using System.ComponentModel;
using System.Drawing;

namespace ListViewComponent {
    public static class ListViewDataGenerator {
        public static object GenerateResourcesCollection(SchedulerStorage storage) {
            return storage.Resources.Items;        
        }

        public static object GenerateListViewAppointmentCollection(AppointmentStorage storage, AppointmentBaseCollection sourceAppointments, DayOfWeek dayOfWeek) {
            BindingList<ListViewAppointment> listViewAppointments = new BindingList<ListViewAppointment>();
            foreach(Appointment appointment in sourceAppointments) {
                listViewAppointments.Add(CreateListViewAppointment(storage, appointment, dayOfWeek));
                
            }
            return listViewAppointments;
        }

        public static ListViewAppointment CreateListViewAppointment(AppointmentStorage storage, Appointment sourceAppointment, DayOfWeek dayOfWeek)
        {
            ListViewAppointment listViewAppointment = new ListViewAppointment();
            listViewAppointment.ListViewDate = sourceAppointment.Start;
           
            listViewAppointment.ListViewSubject = sourceAppointment.Subject;
            listViewAppointment.ListViewRecurrencePattern = RecurrenceInfo.GetDescription(sourceAppointment, dayOfWeek);
            listViewAppointment.ListViewIsRecurring = sourceAppointment.IsRecurring;
            listViewAppointment.ListViewDuration = sourceAppointment.Duration.ToString();
            listViewAppointment.ListViewAppointmentID = sourceAppointment.Id;
            listViewAppointment.ListViewLocation = sourceAppointment.Location;
            listViewAppointment.ListViewStatus = storage.Statuses[sourceAppointment.StatusId]; ;
            listViewAppointment.ListViewLabel = storage.Labels[sourceAppointment.LabelId].Color;
            listViewAppointment.SourceAppointment = sourceAppointment;
            return listViewAppointment;
        }
    }

    // ListView appointment
    public class ListViewAppointment {
        public AppointmentStatus ListViewStatus { get; set; }
        public string ListViewSubject { get; set; }
        public string ListViewDuration { get; set; }
        public string ListViewLocation { get; set; }
        public string ListViewRecurrencePattern { get; set; }
        public bool ListViewIsRecurring { get; set; }
        public DateTime ListViewDate { get; set; }
        public Color ListViewLabel { get; set; }
        public Appointment SourceAppointment { get; set; }
        public object ListViewAppointmentID {
            get;
            set;
        }
    }
}

File: VB/DXApplication5/AgendViewComponent/ListViewDataGenerator.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraScheduler
Imports System.ComponentModel
Imports System.Drawing

Namespace ListViewComponent
	Public NotInheritable Class ListViewDataGenerator
		Private Sub New()
		End Sub
		Public Shared Function GenerateResourcesCollection(ByVal storage As SchedulerStorage) As Object
			Return storage.Resources.Items
		End Function

		Public Shared Function GenerateListViewAppointmentCollection(ByVal storage As AppointmentStorage, ByVal sourceAppointments As AppointmentBaseCollection, ByVal dayOfWeek As DayOfWeek) As Object
			Dim listViewAppointments As New BindingList(Of ListViewAppointment)()
			For Each appointment As Appointment In sourceAppointments
				listViewAppointments.Add(CreateListViewAppointment(storage, appointment, dayOfWeek))

			Next appointment
			Return listViewAppointments
		End Function

		Public Shared Function CreateListViewAppointment(ByVal storage As AppointmentStorage, ByVal sourceAppointment As Appointment, ByVal dayOfWeek As DayOfWeek) As ListViewAppointment
			Dim listViewAppointment As New ListViewAppointment()
			listViewAppointment.ListViewDate = sourceAppointment.Start

			listViewAppointment.ListViewSubject = sourceAppointment.Subject
			listViewAppointment.ListViewRecurrencePattern = RecurrenceInfo.GetDescription(sourceAppointment, dayOfWeek)
			listViewAppointment.ListViewIsRecurring = sourceAppointment.IsRecurring
			listViewAppointment.ListViewDuration = sourceAppointment.Duration.ToString()
			listViewAppointment.ListViewAppointmentID = sourceAppointment.Id
			listViewAppointment.ListViewLocation = sourceAppointment.Location
			listViewAppointment.ListViewStatus = storage.Statuses(sourceAppointment.StatusId)

			listViewAppointment.ListViewLabel = storage.Labels(sourceAppointment.LabelId).Color
			listViewAppointment.SourceAppointment = sourceAppointment
			Return listViewAppointment
		End Function
	End Class

	' ListView appointment
	Public Class ListViewAppointment
		Public Property ListViewStatus() As AppointmentStatus
		Public Property ListViewSubject() As String
		Public Property ListViewDuration() As String
		Public Property ListViewLocation() As String
		Public Property ListViewRecurrencePattern() As String
		Public Property ListViewIsRecurring() As Boolean
		Public Property ListViewDate() As DateTime
		Public Property ListViewLabel() As Color
		Public Property SourceAppointment() As Appointment
		Public Property ListViewAppointmentID() As Object
	End Class
End Namespace
